lib/reservation_mngr.rb

```diff
@@ -0,0 +1,78 @@
+require 'time'
+require 'date'
+
+
+require_relative 'reservation'
+
+module Hotel
+
+  class Reservation_mngr < Reservation
+
+    attr_reader :rooms, :reservations
+    attr_accessor :current_res_id
+
+    def initialize
+      @rooms = build_room_list
+      @reservations = []
+      @current_res_id = 1
```

Comment:
You use `@current_res_id` to store the value of what the next reservation's reservation_id will be. It feels a little weird to me that `Reservation_mngr` hast to hold onto this, and has to have the logic to deal with this and update it and set it, since that feels kind of like a very brittle way to set Reservation ids. Is there a way we can remove this logic from `Reservation_mngr`? Also, does a reservation really need an id?